lib/hsts.c

```diff
@@ -528,8 +528,8 @@ static CURLcode hsts_load(struct hsts *h, const char *file)
       char *lineptr = Curl_dyn_ptr(&buf);
       while(*lineptr && ISBLANK(*lineptr))
         lineptr++;
-      if(*lineptr == '#')
-        /* skip commented lines */
+      if((*lineptr == '#') || strlen(lineptr) == 0)
+        /* skip empty or commented lines */
```

Comment:
Since `Curl_get_line()` works hard to return lines with a trailing newline, should this perhaps rather check for `<= 1` ?

Reply:
That's a good point, it should. Rebased the PR with this fix.